File: node-js/backend/src/models/Channel.ts
import mongoose from 'mongoose';

export interface IChannel extends mongoose.Document {
  _id: mongoose.Types.ObjectId;
  name: string;
  description?: string;
  isPrivate: boolean;
  isDirectMessage: boolean;
  members: mongoose.Types.ObjectId[];
  admins: mongoose.Types.ObjectId[];
  createdBy: mongoose.Types.ObjectId;
  createdAt: Date;
  updatedAt: Date;
}

const channelSchema = new mongoose.Schema<IChannel>(
  {
    name: {
      type: String,
      required: true,
      trim: true,
      maxlength: 50,
    },
    description: {
      type: String,
      trim: true,
      maxlength: 200,
    },
    isPrivate: {
      type: Boolean,
      default: false,
    },
    isDirectMessage: {
      type: Boolean,
      default: false,
    },
    members: [{
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
    }],
    admins: [{
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
    }],
    createdBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
  },
  {
    timestamps: true,
  }
);

// Index for better performance
channelSchema.index({ members: 1 });
channelSchema.index({ createdBy: 1 });
channelSchema.index({ isDirectMessage: 1, members: 1 });

export const Channel = mongoose.model<IChannel>('Channel', channelSchema);

File: backend/dotnet/src/Models/Channel.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;

namespace Backend.Models;

[Collection("channels")]
public class Channel
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

    [Required]
    [StringLength(50)]
    public string Name { get; set; } = string.Empty;

    [StringLength(200)]
    public string? Description { get; set; }

    public bool IsPrivate { get; set; } = false;

    public bool IsDirectMessage { get; set; } = false;

    public List<string> Members { get; set; } = new();

    public List<string> Admins { get; set; } = new();

    [Required]
    public string CreatedBy { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties (populated by separate queries)
    [BsonIgnore]
    public User? CreatedByUser { get; set; }

    [BsonIgnore]
    public List<User> MemberUsers { get; set; } = new();
}